tests/realtikvtest/testkit.go

```diff
@@ -107,25 +99,39 @@ func RunTestMain(m *testing.M) {
 	goleak.VerifyTestMain(testmain.WrapTestingM(m, callback), opts...)
 }
 
-// CreateMockStoreAndSetup return a new kv.Storage.
-func CreateMockStoreAndSetup(t *testing.T, opts ...mockstore.MockTiKVStoreOption) kv.Storage {
-	store, _ := CreateMockStoreAndDomainAndSetup(t, opts...)
-	return store
+type realtikvStoreOption struct {
+	retainData bool
+	keyspace   string
+}
+
+// RealTiKVStoreOption is the config option for creating a real TiKV store.
+type RealTiKVStoreOption func(opt *realtikvStoreOption)
+
+// WithRetainData allows the store to retain old data when creating a new store.
+func WithRetainData() RealTiKVStoreOption {
+	return func(opt *realtikvStoreOption) {
+		opt.retainData = true
+	}
 }
 
-// CreateMockStoreAndDomainAndSetup return a new kv.Storage and *domain.Domain.
-func CreateMockStoreAndDomainAndSetup(t *testing.T, opts ...mockstore.MockTiKVStoreOption) (kv.Storage, *domain.Domain) {
-	if kerneltype.IsNextGen() && *KeyspaceName == "" {
-		// in nextgen kernel, SYSTEM keyspace must be bootstrapped first, if we
-		// don't specify a keyspace which normally is not specified, we use SYSTEM
-		// keyspace as default to make sure test cases can run correctly.
-		*KeyspaceName = keyspace.System
+// WithKeyspaceName allows the store to use a specific keyspace name.
+func WithKeyspaceName(name string) RealTiKVStoreOption {
+	return func(opt *realtikvStoreOption) {
+		opt.keyspace = name
 	}
-	return Setup(t, *KeyspaceName, opts...)
 }
 
-// Setup initializes a kv.Storage and a domain.Domain.
-func Setup(t *testing.T, ks string, opts ...mockstore.MockTiKVStoreOption) (kv.Storage, *domain.Domain) {
+// CreateMockStoreAndSetup return a new kv.Storage.
+func CreateMockStoreAndSetup(t *testing.T, opts ...RealTiKVStoreOption) kv.Storage {
+	store, _ := CreateMockStoreAndDomainAndSetup(t, opts...)
+	return store
+}
+
+// CreateMockStoreAndDomainAndSetup initializes a kv.Storage and a domain.Domain.
+func CreateMockStoreAndDomainAndSetup(t *testing.T, opts ...RealTiKVStoreOption) (kv.Storage, *domain.Domain) {
+	//nolint: errcheck
+	_ = kvstore.Register(config.StoreTypeTiKV, &driver.TiKVDriver{})
```

Comment:
why not check it directly?

Reply:
Because it will report error if this is the second time to register a type.